datafusion/expr/src/built_in_function.rs

```diff
@@ -464,12 +479,16 @@ impl BuiltinScalarFunction {
             },
             BuiltinScalarFunction::ArrayConcat => {
                 let mut expr_type = Null;
+                let mut max_dims = 1;
```

Comment:
If the init value is 1 and all the lists are 1D, we will get null for expr_type, right? But why doesn't the 1D concat test fails?

init value for expr_type = null, and max_dims = 0, with updating expr_type with input_expr_type instead of field since much more straightforward to me.